### DISCUSSION TITLE
Allow to register new commands for `mkdocs` executable

### DISCUSSION THREAD

**Discussion by squidfunk** [mkdocs/mkdocs#3475] [Ideas]
Would it be possible to allow for plugins or other MkDocs related packages to register new commands with the `mkdocs` executable? `git` follows a similar approach, and allows to register subcommands. For example, if a MkDocs extension package would ship an executable that is called:

```
mkdocs-blog-new
```

... it could resolve to:

```
mkdocs blog new
```

This would allow to create a simple CLI to manage blog posts and other interesting applications.

**Comment by Andre601** [comment]
That sounds cool.

Tho, maybe for that case could a sort template system be considered?

Like f.e. you could register a template called `blog` and when a user would use `mkdocs new --template blog` would the template be used that it provides (which in this example would be a basic structure for a blog, including a mkdocs.yml with all required settings)
